### PR TITLE
chore(ln-client): improve logging when choosing gateway

### LINKED ISSUE #4194: Log should tell which gateway was chosen
This message could be improved by giving some details about the gateway that was selected:
https://github.com/fedimint/fedimint/blob/58717cb4a08ec89dc4631251a82e96d06a7b9a9d/modules/fedimint-ln-client/src/lib.rs#L855

### PR DESCRIPTION
**Pull request by elsirion**
Fixes #4194

### CODE REVIEW

File: modules/fedimint-ln-client/src/lib.rs [hunk 1]
```diff
@@ -838,26 +838,29 @@ impl LightningClientModule {
...
                 if !vetted.is_empty() {
-                    debug!("Choosing a vetted gateway");
-                    vetted
+                    let chosen_gateway = vetted
                         .into_iter()
                         .map(|gw| gw.info)
                         .choose(&mut rand::thread_rng())
-                        .ok_or(anyhow::anyhow!("Could not choose a vetted gateway"))
+                        .expect("vetted is non-empty");
+                    debug!(gateway_id = %chosen_gateway.gateway_id, %num_gateways, "Choosing a vetted gateway");
```
Comment (Maan2003) [inline modules/fedimint-ln-client/src/lib.rs:854]: you don't use display for primitive types - https://docs.rs/tracing/latest/tracing/trait.Value.html#foreign-impls
```suggestion
                    debug!(gateway_id = %chosen_gateway.gateway_id, num_gateways, "Choosing a vetted gateway");
```

File: modules/fedimint-ln-client/src/lib.rs [hunk 2]
```diff
@@ -838,26 +838,29 @@ impl LightningClientModule {
...
                     .collect::<Vec<_>>();
                 if !vetted.is_empty() {
-                    debug!("Choosing a vetted gateway");
-                    vetted
+                    let chosen_gateway = vetted
                         .into_iter()
                         .map(|gw| gw.info)
                         .choose(&mut rand::thread_rng())
-                        .ok_or(anyhow::anyhow!("Could not choose a vetted gateway"))
+                        .expect("vetted is non-empty");
```
Comment (15IITian) [inline modules/fedimint-ln-client/src/lib.rs:853]: But this error message  will  log  when there is panic due to  some bug or unexpected condition -> so IMO , this should not be the message.

Comment (Maan2003) [reply modules/fedimint-ln-client/src/lib.rs:853]: feels correct to me, given we just check vetted.is_empty just above

**Review by Maan2003 — APPROVED** [review]

File: modules/fedimint-ln-client/src/lib.rs (hunk 1 above)
Comment (elsirion) [reply modules/fedimint-ln-client/src/lib.rs]: Thx, updated!

**Review by justinmoon — APPROVED** [review]

**Review by dpc — APPROVED** [review]

**Pull request merged**